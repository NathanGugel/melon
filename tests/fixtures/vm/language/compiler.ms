let c = compiler.compile("io.print(5);");

let module = { path = null };

c(module);
c(module);
c = compiler.compile("return 10;");
io.print(c(module));

c = compiler.compile("testtest", |message, file, line, col| => {
    io.print("Error at " .. string.toString(file) .. " (" .. string.toString(line) .. ":" .. string.toString(col) .. ") - " .. message);
});

c = compiler.compile("if(", |message, file, line, col| => {
    io.print("Error at " .. string.toString(file) .. " (" .. string.toString(line) .. ":" .. string.toString(col) .. ") - " .. message);
});

c = compiler.compile("asdasd2", |message, file, line, col| => {
    io.print("Error at " .. string.toString(file) .. " (" .. string.toString(line) .. ":" .. string.toString(col) .. ") - " .. message);
});

c = compiler.compile("{}");
c(module);

c = compiler.compile("return |num| => { return num^2; };");
io.print(c(module)(5));

c = compiler.compile("let a = 5; return |num| => { return a * num; };");
io.print(c(module)(10));

c = compiler.compileFile("wrong_file.ms");
io.print(c);

c = compiler.compileFile("fixtures/vm/language/compiler_test.src");
io.print(c);
c(module).myStuff();

c = compiler.compileFile("fixtures/vm/language/compiler_test_fail.src", |message, file, line, col| => {
    io.print("Error at " .. string.toString(file) .. " (" .. string.toString(line) .. ":" .. string.toString(col) .. ") - " .. message);
});