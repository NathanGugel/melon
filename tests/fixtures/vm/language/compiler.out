5
5
10
Error at null (1:7) - Invalid statement. Function call or variable assignment expected.
Error at null (1:2) - Call or assignment expected
Error at null (1:2) - Expected token 'RIGHT_PAREN' but found 'EOF'
Error at null (1:2) - Call or assignment expected
Error at null (1:6) - Invalid statement. Function call or variable assignment expected.
25
50
null
[Closure main]
Working: 10
Error at fixtures/vm/language/compiler_test_fail.src (1:6) - Invalid statement. Function call or variable assignment expected.
